src/components/TaskItem/Drawer/Drawer.tsx

```diff
@@ -40,104 +42,151 @@ export function TaskItemDrawer({
 }: TaskItemDrawerProps): ReactElement {
   const { enqueueSnackbar } = useSnackbar()
   const { mutateAsync } = useUpdateProcess()
+  const [taskStatus, setTaskStatus] = useState(task.status)
+  const [postType, setPostType] = useState<string>()
 
-  function onClick(status: Task['status']): () => Promise<void> {
-    return async function (): Promise<void> {
-      const taskLists = process.taskLists
-      const taskIndex = findIndex(taskList.tasks, ['_id', task._id])
-      taskList.tasks[taskIndex] = { ...task, status }
-      const taskListIndex = findIndex(taskLists, ['title', taskList.title])
-      taskLists[taskListIndex] = taskList
+  async function handleSubmit(): Promise<void> {
```

Comment:
```suggestion
  async function handleSubmit(status = taskStatus): Promise<void> {
```

this will allow you to set status by arg otherwise use the taskStatus.